idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtGenerateMembersHandler.kt

```diff
@@ -0,0 +1,284 @@
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.analyse
+import org.jetbrains.kotlin.idea.frontend.api.components.KtDeclarationRendererOptions
+import org.jetbrains.kotlin.idea.frontend.api.components.KtTypeRendererOptions
+import org.jetbrains.kotlin.idea.frontend.api.components.RendererModifier
+import org.jetbrains.kotlin.idea.frontend.api.components.ShortenCommand
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtCallableSymbol
+import org.jetbrains.kotlin.idea.frontend.api.tokens.HackToForceAllowRunningAnalyzeOnEDT
+import org.jetbrains.kotlin.idea.frontend.api.tokens.hackyAllowRunningOnEdt
+import org.jetbrains.kotlin.idea.util.application.runWriteAction
+import org.jetbrains.kotlin.psi.KtCallableDeclaration
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.prevSiblingOfSameType
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.utils.addToStdlib.firstIsInstanceOrNull
+
+internal abstract class KtGenerateMembersHandler : AbstractGenerateMembersHandler<KtClassMember>() {
+    @OptIn(HackToForceAllowRunningAnalyzeOnEDT::class)
+    override fun generateMembers(
+        editor: Editor,
+        classOrObject: KtClassOrObject,
+        selectedElements: Collection<KtClassMember>,
+        copyDoc: Boolean
+    ) {
+        // Using hackyAllowRunningOnEdt here because we don't want to pre-populate all possible textual overrides before user selection.
+        val commands = hackyAllowRunningOnEdt {
+            val batches = analyse(classOrObject) {
+                this.generateMembers(editor, classOrObject, selectedElements, copyDoc)
+            }
+            // Reference shortening is done in a separate analysis session because the session need to be aware of the newly generated
+            // members.
+            analyse(classOrObject) {
+                batches.map { batch ->
+                    val first = batch.first()
+                    val last = batch.last()
+                    collectPossibleReferenceShortenings(first.containingKtFile, TextRange(first.startOffset, last.endOffset))
+                }
+            }
+        }
+        runWriteAction {
+            commands.forEach { it.invokeShortening() }
+        }
+    }
+
+    /**
+     * @returns the generated PSI element inside the target class. The members are grouped together if they are near each other. Each group contains the members in the order they appear in the target class after code generation
+     *
+     */
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun KtAnalysisSession.generateMembers(
+        editor: Editor,
+        currentClass: KtClassOrObject,
+        selectedElements: Collection<KtClassMember>,
+        copyDoc: Boolean
+    ): List<Iterable<KtDeclaration>> {
```

Comment:
Please, encapsulate this result into class, to undersatand what that function returns a matrix of declarations to be shortened